File: Python3_PyPy3/q_table.py
from typing import List


class QTable:
    def __init__(self, state_size: int, action_size: int, init_qvalue: float):
        self.state_size = state_size
        self.action_size = action_size
        self.init_qvalue = init_qvalue
        self.table = [[init_qvalue] * action_size for _ in range(state_size)]

File: Ruby_MRI_JIT/q_table.rb
class QTable
  attr_reader :state_size, :action_size, :init_qvalue
  attr_accessor :table


  def initialize(state_size, action_size, init_qvalue)
    @state_size = state_size
    @action_size = action_size
    @init_qvalue = init_qvalue
    @table = Array.new(state_size) {
      Array.new(action_size, @init_qvalue)
    }
  end
end